src/http/client.cr

```diff
@@ -661,8 +661,12 @@ class HTTP::Client
   private def send_request(request)
     decompress = set_defaults request
     run_before_request_callbacks(request)
-    request.to_io(io)
-    io.flush
+    begin
+      request.to_io(io)
+      io.flush
+    rescue e : IO::Error
+      raise e unless e.os_error == Errno::EPIPE
```

Comment:
So if writing the request fails because of a broken pipe, it's just silently ignored?

Reply:
You know, that's a good point. I was only thinking about it in terms of a failed initial write.
.. So did it ignore the error in the past? Hmm.

Perhaps a different approach is needed then. Or at least another flag to ignore the error for the first write, similar to the 'flush' call